src/events/jsonEvent.ts

```diff
@@ -2,39 +2,39 @@ import { v4 as uuid } from "uuid";
 
 export type JSONType = Record<string | number, unknown> | unknown[];
 
-export interface JSONEventData {
+export interface JSONEventData<T extends string = string, D extends JSONType = JSONType, M extends JSONType = JSONType> {
```

Comment:
Can we fully name these: `Type`, `Data`, `Metadata`. 

Just makes it a bit clearer for using:

![image](https://user-images.githubusercontent.com/11861797/103176080-402a0f00-486f-11eb-9911-69988455dd3e.png)

Reply:
Agree with that, changed in 4bcb16755e195d48d4959cb7b69987b7cbdd3cda